core-test/src/test/scala/org/bitcoins/core/protocol/ln/currency/MilliSatoshisTest.scala

```diff
@@ -28,6 +25,63 @@ class MilliSatoshisTest extends FlatSpec with MustMatchers {
     MilliSatoshis.fromPico(PicoBitcoins(110)) must be(MilliSatoshis(11))
   }
 
+  it must "add millisatoshis" in {
+    PropertyChecks.forAll(LnCurrencyUnitGen.milliSatoshisPair) {
+      case (first, second) =>
+        val bigInt = first.toBigInt + second.toBigInt
+        assert((first + second).toBigInt == bigInt)
+    }
+  }
+
+  private val msatWithNum = for {
+    msat <- LnCurrencyUnitGen.milliSatoshis
+    num <- NumberGenerator.bigIntsUInt64Range.filter(_ > 0)
+  } yield (msat, num)
+
+  it must "multiply millisatoshis with an int" in {
+    PropertyChecks.forAll(msatWithNum) {
+      case (msat, bigint) =>
+        val underlyingCalc = msat.toBigInt * bigint
+        assert((msat * bigint).toBigInt == underlyingCalc)
```

Comment:
shouldn't this hit an assertion if `msat * bigInt` results in too big of a number? I forget if `msat` is bound to a protocol type (UInt32, UInt64 etc)

Reply:
msats are only lower bound IIRC, with bigint as underlying